### PR TITLE
[TEST] ASoC: rt700: disable jack detection when the codec suspends

### PR DESCRIPTION
**Pull request by shumingfan**
This patch tries to avoid the unexpected interrupt to do the jack detection after the codec suspends.

TEST only for #2943 

Signed-off-by: Shuming Fan <shumingf@realtek.com>

### CODE REVIEW

File: sound/soc/codecs/rt700-sdw.c [hunk 1]
```diff
@@ -517,6 +535,23 @@ static int __maybe_unused rt700_dev_resume(struct device *dev)
...
+			RT700_SET_MIC2_UNSOLICITED_ENABLE, 0x82);
+		regmap_write(rt700->regmap,
+			RT700_SET_HP_UNSOLICITED_ENABLE, 0x81);
+		regmap_write(rt700->regmap,
+			RT700_SET_INLINE_UNSOLICITED_ENABLE, 0x83);
+
+		dev_dbg(dev, "in %s jd enable\n", __func__);
+
+		regmap_write(rt700->regmap,
+			RT700_SET_AUDIO_POWER_STATE, AC_PWRST_D3);
```
Comment (lyakh) [inline sound/soc/codecs/rt700-sdw.c:552]: I guess this is correct somehow and has been tested, but just seems a bit odd... Is it really switching to D3 during resume?

Comment (shumingfan) [reply sound/soc/codecs/rt700-sdw.c:552]: It is just the power-saving mode for the codec. The codec driver should go to power-saving mode before going to the suspend state. So, I just set back to power-saving mode during the resume function.

Comment (lyakh) [reply sound/soc/codecs/rt700-sdw.c:552]: ok, but we're resuming for playback, the codec goes to D3 here and then presumably it's switched back to D0 immediately, right? Is that common or has some specific reason? Naively I'd expect `.resume()` to bring the system to a fully operational mode to be able to begin operation immediately. Although bringing the hardware to some intermediate state, waiting for the actual operation to begin, should work too...

**Comment by plbossart** [comment]
@shumingfan @bardliao  I am not sure if we want this 'eature'. The Intel parts are capable of detecting a SoundWire wake while in pm_runtime suspend. I am not sure why we would disable this capability at the codec level. It's fine for a test, but it doesn't look like a good move.

**Comment by shumingfan** [comment]
> @shumingfan @bardliao I am not sure if we want this 'eature'. The Intel parts are capable of detecting a SoundWire wake while in pm_runtime suspend. I am not sure why we would disable this capability at the codec level. It's fine for a test, but it doesn't look like a good move.

@plbossart This patch just is used to debug and test for issue #2943. After the testing, this patch can't fix the issue and the problem should be why the codec driver goes to pm_runtime suspend from the resume so quick.
We will check the issue first. After that, I will close this PR.

**Pull request closed without merging**